### PR TITLE
[CS2103-W15-4] Pet Store Helper

### PR DESCRIPTION
**Pull request by umaikaze**

### CODE REVIEW

**Comment by umaikaze** [comment]
Usernames of team members: @shenghaoc @tianqi77 @Zhu-Ruicong @limdylan

**Comment by raivatshah** [comment]
I like how Model was edited from the default version with specific changes. It is more clear with the Model to understand the codebase.

You may find it helpful to draw class diagrams before you implement all classes as it will allow all members of the team to come to a clear consensus about the functionality and structure of each class.

**Comment by harrychengly** [comment]
I think it would be good to include some diagrams in the implementations section when explaining a feature. It would be helpful for the developer to visualise the flow of the feature when reading.

**Comment by terencemah** [comment]
Some of the classes mentioned in the Developer Guide are not introduced before, so it is not clear what they refer to. Subsequently, it would be good to introduce all relevant classes systematically, so that there is coherent flow.

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -142,105 +138,164 @@ image::StorageClassDiagram.png[]
...
 
-[NOTE]
-If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+Step 4. The user executes `display p` to view the list of **p**ets. The `display` command calls `Model#changeDisplaySystem`, which causes `filteredDisplayItems` to re-binded to `filteredPets`. Since the program should now display a list of items instead of a calendar, upon receiving back the result of the command in `MainWindow`, `resultDisplayPlaceholder` is changed to display the `DisplayListPanel` from `CalendarPanel`.
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+The following \____ diagram shows how the display operation works: (WIP)
 
-image::UndoRedoState3.png[]
+image::displayDiagram2.png[]
```
Comment (chenjiangw) [inline docs/DeveloperGuide.adoc]: displayDiagram2 not showing.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -142,105 +138,164 @@ image::StorageClassDiagram.png[]
...
+- In the `MainWindow`, a `DisplayListPanel` and a `CalendarPanel` are initialized. The `DisplayListPanel` is created with `filteredDisplayItems` as its backing list, while the `CalendarPanel` will be initialized with the `List` of `Slot`s obtained from `logic.getPetTracker().getSlotList()`.
 
-image::UndoRedoState0.png[]
+- The `resultDisplayPlaceholder` component in the `MainWindow` is set to display the `DisplayListPanel`. As such, the user sees the list of pets as the initial display.
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+The following \____ diagram summarises Step 1 and shows what happens at initialization: (WIP)
 
-image::UndoRedoState1.png[]
+image::displayDiagram1.png[]
```
Comment (chenjiangw) [inline docs/DeveloperGuide.adoc]: displayDiagram1 not showing.

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -142,105 +138,164 @@ image::StorageClassDiagram.png[]
...
 == Implementation
 
 This section describes some noteworthy details on how certain features are implemented.
 
-// tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+// tag::display[]
+=== Display
+==== Implementation
```
Comment (chenjiangw) [inline docs/DeveloperGuide.adoc:238]: Can include an activity diagram to facilitate understanding.

**Comment by PotatoCombat** [comment]
1. (3.1.4) You should use a class diagram to explain it
2. Step 2 should be broken down. It is too wordy and complicated.
3. The ! thing should be after step 3
4. In general, break down your steps into smaller simpler steps.
5. Section 3.6 - you don’t explain how it’s done.
